File: sandbox/java/src/main/java/com/dropbox/sign_sandbox/FaxDraftCreateExample.java
package com.dropbox.sign_sandbox;

import com.dropbox.sign.ApiException;
import com.dropbox.sign.Configuration;
import com.dropbox.sign.api.FaxApi;
import com.dropbox.sign.auth.HttpBasicAuth;
import com.dropbox.sign.model.FaxDraftCreateRequest;
import com.dropbox.sign.model.SubEditorPageOptions;

import java.util.List;

public class FaxDraftCreateExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((HttpBasicAuth) config.getAuthentication("api_key")).setUsername("YOUR_API_KEY");

        var editorOptions = new SubEditorPageOptions();
        editorOptions.forceUploadPage(false);
        editorOptions.forceEditorPage(true);
        editorOptions.forceReviewPage(true);

        var faxDraftCreateRequest = new FaxDraftCreateRequest();
        faxDraftCreateRequest.clientId("b6b8e7deaf8f0b95c029dca049356d4a2cf9710a");
        faxDraftCreateRequest.fileUrls(List.of(
            "https://www.dropbox.com/s/ad9qnhbrjjn64tu/mutual-NDA-example.pdf?dl=1"
        ));
        faxDraftCreateRequest.recipients(List.of("+14155552671"));
        faxDraftCreateRequest.editorOptions(editorOptions);
        faxDraftCreateRequest.testMode(true);

        try
        {
            var response = new FaxApi(config).faxDraftCreate(
                faxDraftCreateRequest
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling FaxApi#faxDraftCreate");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: examples/FaxDraftCreateExample.rb
require "json"
require "dropbox-sign"

Dropbox::Sign.configure do |config|
    config.username = "YOUR_API_KEY"
end

editor_options = Dropbox::Sign::SubEditorPageOptions.new
editor_options.force_upload_page = false
editor_options.force_editor_page = true
editor_options.force_review_page = true

fax_draft_create_request = Dropbox::Sign::FaxDraftCreateRequest.new
fax_draft_create_request.client_id = "b6b8e7deaf8f0b95c029dca049356d4a2cf9710a"
fax_draft_create_request.file_urls = [
    "https://www.dropbox.com/s/ad9qnhbrjjn64tu/mutual-NDA-example.pdf?dl=1",
]
fax_draft_create_request.recipients = ["+14155552671"]
fax_draft_create_request.editor_options = editor_options
fax_draft_create_request.test_mode = true

begin
    response = Dropbox::Sign::FaxApi.new.fax_draft_create(
        fax_draft_create_request,
    )

    p response
rescue Dropbox::Sign::ApiError => e
    puts "Exception when calling FaxApi#fax_draft_create: #{e}"
end
